### PR TITLE
refactor(core): dump AnyMessage as a struct

### PR DESCRIPTION
**Pull request by GoldsteinE**
Any other serialization mode still uses a tuple representation for compactness. Using struct in the dumping mode makes it more readable.

### CODE REVIEW

File: elfo-core/src/message.rs [hunk 1]
```diff
@@ -150,15 +150,21 @@ impl Serialize for AnyMessage {
...
-        tuple.serialize_element(self.protocol())?;
-        tuple.serialize_element(self.name())?;
-
         // TODO: avoid allocation here
         let erased_msg = self._erase();
-        tuple.serialize_element(&*erased_msg)?;
-
-        tuple.end()
+        if crate::scope::serde_mode() == SerdeMode::Dumping {
+            let mut fields = serializer.serialize_struct("RobotMessage", 3)?;
```
Comment (jazvit) [inline elfo-core/src/message.rs:156]: Message/AnyMessage

**Review by jazvit — APPROVED** [review]
LGTM

File: elfo-core/src/message.rs [hunk 2]
```diff
@@ -150,15 +150,21 @@ impl Serialize for AnyMessage {
...
-        let mut tuple = serializer.serialize_tuple(3)?;
-        tuple.serialize_element(self.protocol())?;
-        tuple.serialize_element(self.name())?;
-
         // TODO: avoid allocation here
         let erased_msg = self._erase();
-        tuple.serialize_element(&*erased_msg)?;
-
-        tuple.end()
+        if crate::scope::serde_mode() == SerdeMode::Dumping {
```
Comment (loyd) [inline elfo-core/src/message.rs:155]: We should comment that `Serialize` and `Deserialize` instances run only if `AnyMessage` is used inside other messages (although it's unstable anyway).

File: elfo-core/src/message.rs (hunk 1 above)
Comment (loyd) [reply elfo-core/src/message.rs:156]: `AnyMessage`. Messages are encoded differently.

**Review by loyd — APPROVED** [review]

**Pull request merged**